Src/FluentAssertions/Collections/SelfReferencingCollectionAssertions.cs

```diff
@@ -647,34 +647,34 @@ public AndConstraint<TAssertions> NotContain(Expression<Func<T, bool>> predicate
             Guard.ThrowIfArgumentIsNull(predicate, nameof(predicate));
 
             string expectationPrefix =
-                string.Format("Expected {{context:collection}} to contain a single item matching {0}{{reason}}, ", predicate.Body);
+                "Expected {context:collection} to contain a single item matching {0}{reason}, ";
```

Comment:
@jnyrup what's wrong with:

```csharp
string expectationPrefix =
                $"Expected {{context:collection}} to contain a single item matching {predicate.Body.ToString().EscapePlaceholders()}{{reason}}, ";
```

Reply:
JFYI guys, I personally support the idea not to format the string twice. So the current approach looks better for me.

Reply:
I get that, but this is what that version gives me:

```Expected collection to contain a single item matching (item == "{123}"), but no such item was found.```

Reply:
Using `EscapePlaceholders` will work in many cases, but e.g. this tests fails with `EscapePlaceholders()` but passes with the current PR changes.
```cs
// Arrange
IEnumerable<string> collection = null;

// Act
Action act = () => collection.Should().ContainSingle(item => item == "{reason}");

// Assert
act.Should().Throw<XunitException>().WithMessage("*{reason}*");
```

It's equivalent to keeping query and input separated in SQL statements.